seed/scripts/main.js

```diff
@@ -0,0 +1,166 @@
+    }
+  },
+  log (message){
+    return (data) => {
+      console.log(message)
+      return data
+    }
+  }
+}
+
+const Async = {
+  createPromise (module) {
+    return Promise
+    .resolve()
+    .catch(Logger.debug(module))
+  },
+  createQueue (fns, promise) {
+    return fns.reduce((p, fn) => {
+      return p.then(fn)
+    }, promise)
+  }
+}
+
+const DB = {
+  project: {name: 'project', mapper: ProjectMapper},
+  tech: {name: 'tech', mapper: TechMapper},
+  store (item, db) {
+    return db.mapper
+    .create(item)
+    .catch(`${db.name}Store`)
+  },
+  find (query, db) {
+    return db.mapper
+    .findAll(query)
+    .then(items => items.length > 0 ? items[0] : null)
+    .catch(`${db.name}Find`)
+  }
+}
+
+const GH = {
+  getFullName (link) {
+    const path = link.split('github.com/')[1]
+    const [owner, repo] = path.split('/')
+    if (owner && repo) {
+      return `${owner}/${repo}`
+    }else{
+      throw `Cannot get fullName from: ${link}`
+    }
+  },
+  search (fullName) {
+    const search = new GitHub().search()
+
+    return search
+    ._request('GET', `https://api.github.com/repos/${fullName}`)
+    .catch(Logger.debug('github'))
+    .then(result => result.data)
+  }
+}
+
+const Tech = {
+  createQuery (tech) {
+    return () => {
+      return {where: {id: {'===': tech.id}}}
+    }
+  },
+  createTask (tech, position, total) {
+    const id = tech.id
+
+    return Async
+    .createPromise(tech)
+    .then(Logger.log(`Tech: ${position+1}/${total}`))
+    .then(Logger.log(`${id}: searching on DB`))
+    .then(Tech.createQuery(tech))
+    .then((query) => {
+      return DB.find(query, DB.tech)
+    })
+    .then(savedTech => {
+      if (savedTech) {
+        throw 'is already saved'
+      }
+    })
+    .then(Logger.log('saving on DB'))
+    .then(() => {
+      DB.store(tech, DB.tech)
+    })
+    .then(Logger.log('saved'))
+    .catch(Logger.debug(id))
+  }
+}
+
+const Project = {
+  createQuery (fullName) {
+    return {where: {full_name: {'===': fullName}}}
+  },
+  createTask (project, position, total) {
+    let fullName = ''
+
+    return Async
+    .createPromise()
+    .then(() => {
+      fullName = GH.getFullName(project.link)
+    })
+    .then(Logger.log(`Project: ${position+1}/${total}`))
+    .then(Logger.log('searching on DB'))
+    .then(Project.createQuery(fullName))
+    .then((query) => {
+      return DB.find(query, DB.project)
+    })
+    .then(savedProject => {
+      if (savedProject) {
+        throw 'is already saved'
+      }
+    })
+    .then(Logger.log('searching on Github'))
+    .then(() => {
+      return GH.search(fullName)
+    })
+    .then(Logger.log('saving on DB'))
+    .then((githubData) => {
+      DB.store({...githubData, tech: project.tech}, DB.project)
+    })
+    .then(Logger.log('saved'))
+    .catch(Logger.debug('Project'))
+  }
+}
+
+
+const tasks = Async.createPromise('tasks')
+
+const techTasks = techs.reduce((previous, tech, index, array) => {
```

Comment:
This and the reduce below it should be a map. A reduce that gets an array and return an array is a map. Reduce should be used to convert an array to a single value.